### ISSUE TITLE
ST_Transform sometimes returns POINT(Infinity Infinity)

### ISSUE THREAD

**Issue by jc9677** [duckdblabs/duckdb_spatial#119]
Here's a simple example. The first point gets transformed properly, but the second one returns POINT (Infinity Infinity), even though only one digit has been changed:

```
import duckdb as ddb
ddb.load_extension('spatial')

# Create a table with columns for Longitude and Latitude
ddb.sql("CREATE OR REPLACE TABLE test (Longitude DOUBLE, Latitude DOUBLE)")

# Insert a row with a point
ddb.sql("INSERT INTO test VALUES (-123.12345, 49.12345)")
ddb.sql("INSERT INTO test VALUES (-123.12345, 49.02345)")


ddb.sql("SELECT *, ST_Point(Longitude, Latitude) AS GEOM, ST_Transform(ST_FlipCoordinates(GEOM), 'EPSG:4326', 'ESRI:102001') AS AEAC FROM test")
```
OUTPUT:

```
┌────────────┬──────────┬─────────────────────────────┬─────────────────────────────────────────────┐
│ Longitude  │ Latitude │            GEOM             │                    AEAC                     │
│   double   │  double  │          geometry           │                  geometry                   │
├────────────┼──────────┼─────────────────────────────┼─────────────────────────────────────────────┤
│ -123.12345 │ 49.12345 │ POINT (-123.12345 49.12345) │ POINT (-1930488.849258097 1393621.32509929) │
│ -123.12345 │ 49.02345 │ POINT (-123.12345 49.02345) │ POINT (Infinity Infinity)                   │
└────────────┴──────────┴─────────────────────────────┴─────────────────────────────────────────────┘
```

I assume this is related to how PROJ decides which coordinate order to use. If I manually use PROJ to transform the second point, it fails in the same way unless I set **always_xy** to True.
This fails:
```
from pyproj import CRS, Transformer

# Create a transformer from WGS84 to ESRI:102001
transformer = Transformer.from_crs("EPSG:4326", "ESRI:102001")

# Transform the point
transformer.transform(-123.12345, 49.02345)
```

```
OUTPUT:
(inf, inf)
```

But with always_xy=True, it works:
```
from pyproj import CRS, Transformer

# Create a transformer from WGS84 to ESRI:102001
transformer = Transformer.from_crs("EPSG:4326", "ESRI:102001", always_xy=True)

# Transform the point
transformer.transform(-123.12345, 49.02345)
```

```
OUTPUT:
(-1934848.2010468745, 1383417.2910669951)
```

**Comment by Maxxen** [comment]
Hi! Thanks for raising this issue! Im not exactly sure why PROJ acts this way, but I guess a good solution would be to expose the equivalent of `always_xy` as an optional argument to `st_transform`. I'm planning to spend some more time with spatial next week to try to fix some bugs before the next release, I'll try to squeeze this in there as well.

**Comment by jc9677** [comment]
I had tried using always_xy, hoping that it was an undocumented parameter or something... Glad to hear that it might be possible to add it. Thanks for the reply, and for the spatial extension in general. I'm using it a lot!

**Comment by Maxxen** [comment]
I can't really seem to reproduce this but I've added a test case with the steps you provided to ensure this is not a platform related issue, we'll see if the CI manages to reproduce it.

That said I've also added an optional bool parameter on my fork to force PROJ to use the "conventional GIS ordering" (long-lat), which also turns out to be what PostGIS does by default. Will get merged before the release. 

Although it is somewhat convenient to stay unopinionated when it comes to the axis order we might want to take a stance on this as well. As mentioned in #63 it's probably what people expect.
